redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt

```diff
@@ -55,35 +60,51 @@ internal class Items<VH : RecyclerView.ViewHolder>(
   }
 }
 
-internal class ViewLazyList(
-  override val value: RecyclerView,
+/**
+ * Public function to allow downstream factories to create their own LazyList
+ */
+public fun ViewLazyList(
+  recyclerViewFactory: () -> RecyclerView,
+): LazyList<View> = ViewLazyListImpl(recyclerViewFactory)
```

Comment:
I'm a bit confused by this bit. Surely if someone didn't want to use our implementation of `ViewLazyList`, then they could just not depend on the module.